src/QsCompiler/LanguageServer/EditorState.cs

```diff
@@ -157,13 +157,13 @@ internal bool QsProjectLoader(Uri projectFile, out ProjectInformation info)
             return true;
         }
 
-        internal bool QsTemporaryProjectLoader(Uri sourceFileUri, string sdkVersion, out Uri projectUri, out ProjectInformation info)
+        internal Uri QsTemporaryProjectLoader(Uri sourceFileUri, string sdkVersion)
         {
-            projectUri = null;
-            info = null;
-
             var localFolderPath = Path.GetDirectoryName(sourceFileUri.LocalPath);
-            var projectFolderPath = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "qsharp", $"{localFolderPath.GetHashCode():x8}")).FullName;
+            var projectFolderPath = Directory.CreateDirectory(Path.Combine(
+                Path.GetTempPath(),
+                "qsharp",
+                string.Join("_", localFolderPath.Split(Path.GetInvalidFileNameChars())))).FullName;
```

Comment:
Having just dealt with a bunch of edge case bugs related to naming conflicts; what is the reason to not just keep the folder structure relative to the workspace folder? I.e. if the workspace folder is /foo, and the file is /foo/subfolder/file.qs, then why not create a subfolder in the temp folder such that the project file is in tempfolder/subfolder/projFile.csproj?

Reply:
What if a folder is open in multiple VS Code instances with different workspace roots? For example, if I open folder `/foo` in one instance of VS Code, and I open folder `/foo/subfolder` in another instance. Then `/foo/subfolder` has two temporary projects lying around. I'm not sure if that would actually cause any issue (I suspect not, in that case), but wouldn't it be more desirable to have a 1:1 mapping between folder and temporary .csproj?

Reply:
I am not sure I see the benefits, since with two VS Code instances there would also be two servers running, each one compiling one project (the additional file doesn't bother me). 
In any case, there is also another way though to avoid naming conflicts without preserving a hierarchy inspired by url schemes, as pointed out by @samarsha: you could map both the path combinator and the underscore; e.g.  _ -> _x and / -> _y where x and y can be anything but need to be nontrivial.

Reply:
Yes, that approach makes sense. (I know it doesn't really matter much, but the additional file would bother me a little bit; I really don't want to leave more files than necessary on the user's machine. 😊)